src/screens/claimables/ClaimingClaimableSharedUI.tsx

```diff
@@ -0,0 +1,287 @@
+import React, { useEffect, useMemo } from 'react';
+import { AccentColorProvider, Bleed, Box, Inline, Text, TextShadow, globalColors, useColorMode } from '@/design-system';
+import * as i18n from '@/languages';
+import { ListHeader, Panel, TapToDismiss, controlPanelStyles } from '@/components/SmoothPager/ListPanel';
+import { deviceUtils, safeAreaInsetValues } from '@/utils';
+import { View } from 'react-native';
+import { IS_IOS } from '@/env';
+import { ButtonPressAnimation, ShimmerAnimation } from '@/components/animations';
+import { SponsoredClaimable, TransactionClaimable } from '@/resources/addys/claimables/types';
+import RainbowCoinIcon from '@/components/coin-icon/RainbowCoinIcon';
+import { useTheme } from '@/theme';
+import { FasterImageView } from '@candlefinance/faster-image';
+import { chainsLabel } from '@/chains';
+import { useNavigation } from '@/navigation';
+import { TextColor } from '@/design-system/color/palettes';
+import Animated, { useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated';
+import { convertAmountToNativeDisplayWorklet, handleSignificantDecimalsWithThreshold } from '@/__swaps__/utils/numbers';
+import { useAccountSettings } from '@/hooks';
+
+const BUTTON_WIDTH = deviceUtils.dimensions.width - 52;
+
+export type ClaimStatus =
+  | 'idle' // initial state
+  | 'claiming' // user has pressed the claim button
+  | 'pending' // claim has been submitted but we don't have a tx hash
+  | 'success' // claim has been submitted and we have a tx hash
+  | 'error'; // claim has failed
```

Comment:
I get what it is because of the comments but can't we have the status itself say better what it is without the need to explain in comments?

Reply:
i think it's clear enough tbh